thrust: add 1.17 package versions

mpokorny opened the pull request:
Note that the new package versions use the CMake build system, which provides the canonical Thrust installation (replacing the simple copying of source directories to the installation directory in prior versions).

I did not add versions in the v2 series, as I have no need for those versions currently. I don't know much about Thrust, but have a need to use v1.17 as a dependency of another package. I suppose that the newer versions could be added similarly using the CMake build system. I tested installations of v1.17.{0,1,2} and also v1.16.0.

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

alalazo reviewed (commented):
Indentation seems wrong to me. Builder classes must be defined at the module level, not nested into a package.

mpokorny committed:
thrust: add 1.17 package versions

Note that the new package versions use the CMake build system, which
provides the canonical Thrust installation (replacing the simple copying
of source directories to the installation directory in prior versions).

mpokorny commented:
Indentation fixed

alalazo commented on var/spack/repos/builtin/packages/thrust/package.py:
```diff
@@ -45,7 +52,35 @@ class Thrust(Package):
+    depends_on("cmake@3.15:", when="@1.17:", type="build")
+
+
+# CMake provides an 'install' target, which provides the canonical
+# installation for Thrust (and CUB)
```
Can we remove this:
```suggestion
```
? It's a redundant comment:

mpokorny replied:
done

alalazo commented on var/spack/repos/builtin/packages/thrust/package.py:
```diff
@@ -45,7 +52,35 @@ class Thrust(Package):
+        ]
+
+
+# the generic install method for the package was implemented for
+# versions strictly before 1.17, and is maintained for
+# compatibility
```
Similarly:
```suggestion
```

mpokorny replied:
done

alalazo commented on var/spack/repos/builtin/packages/thrust/package.py:
```diff
@@ -45,7 +52,35 @@ class Thrust(Package):
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    def cmake_args(self):
+        return [
+            "-DTHRUST_ENABLE_HEADER_TESTING=OFF",
+            "-DTHRUST_ENABLE_TESTING=OFF",
+            "-DTHRUST_ENABLE_EXAMPLES=OFF",
+            "-DTHRUST_ENABLE_BENCHMARKS=OFF",
+            "-DTHRUST_INCLUDE_CUB_CMAKE=OFF",
```
Here it would be more idiomatic to use `self.define`

mpokorny replied:
done

alalazo reviewed (changes requested):
It seems `thrust` has been discontinued upstream, and is now part of https://github.com/nvidia/cccl 

@fspiga @mpokorny Does it make sense to add a package for `cccl` and deprecate this PR? Or is this possibly still needed?

mpokorny commented:
One problem with deprecating this PR is you'd be left with no way to install `thrust` for older versions that precede `nccl`. It seems like a good reason not to extend this package to v2, however.

alalazo commented:
@spackbot run pipeline

mpokorny commented:
I updated the branch by rebasing with `develop` to get the pipeline to run again (one job had failed for some reason incomprehensible to me), and that appears to have broken the auto-merge. It looks like I should have done a merge update, instead, or maybe just used spackbot to run the pipeline again. @alalazo , would you please merge this PR or re-enable the auto-merge? Sorry about that.
